### ISSUE TITLE
Permission issue when using runIdeForUiTests

### ISSUE THREAD

**Issue by henryju** [JetBrains/gradle-intellij-plugin#681]
Hi,

I'm trying to migrate our plugin to build with version 1.0.

There is now an issue on our CI, that seems to be caused by a permission issue when extracting IDE archive in the Gradle cache, at least for CLion:

`./gradlew runIdeForUiTests`
```
2021-05-27T16:48:57.4719018Z 2021-05-27 16:45:35,700 [ 341778]   WARN -        #com.jetbrains.cidr.cpp - Execution problem: 
2021-05-27T16:48:57.4720523Z com.intellij.execution.process.ProcessNotCreatedException: Cannot run program "/home/vsts/work/1/.gradle/caches/modules-2/files-2.1/com.jetbrains.intellij.clion/clion/2021.1/7e6d797e2e496c7163f91c78bb7c6a55725df6da/clion-2021.1/bin/cmake/linux/bin/cmake": error=13, Permission denied
2021-05-27T16:48:57.4721901Z 	at com.intellij.execution.configurations.GeneralCommandLine.createProcess(GeneralCommandLine.java:385)
2021-05-27T16:48:57.4722763Z 	at com.intellij.execution.process.OSProcessHandler.startProcess(OSProcessHandler.java:100)
2021-05-27T16:48:57.4723567Z 	at com.intellij.execution.process.OSProcessHandler.<init>(OSProcessHandler.java:48)
2021-05-27T16:48:57.4724384Z 	at com.intellij.execution.process.CapturingProcessHandler.<init>(CapturingProcessHandler.java:24)
2021-05-27T16:48:57.4725238Z 	at com.jetbrains.cidr.system.LocalHost$ParametrizedCapturingProcessHandler.<init>(LocalHost.java:173)
2021-05-27T16:48:57.4726031Z 	at com.jetbrains.cidr.system.LocalHost$1.build(LocalHost.java:86)
2021-05-27T16:48:57.4726768Z 	at com.jetbrains.cidr.system.CidrProcessBuilder.build(CidrProcessBuilder.kt:34)
2021-05-27T16:48:57.4727507Z 	at com.jetbrains.cidr.system.LocalHost.runProcess(LocalHost.java:57)
2021-05-27T16:48:57.4728250Z 	at com.jetbrains.cidr.cpp.toolchains.CMakeExecutableTool.executeWithOutput(CMakeExecutableTool.java:143)
2021-05-27T16:48:57.4729068Z 	at com.jetbrains.cidr.cpp.toolchains.CMakeExecutableTool.readVersion(CMakeExecutableTool.java:120)
2021-05-27T16:48:57.4729944Z 	at com.jetbrains.cidr.cpp.toolchains.CMakeExecutableTool.doWithToolchainCMake(CMakeExecutableTool.java:208)
2021-05-27T16:48:57.4730703Z 	at com.jetbrains.cidr.cpp.toolchains.CMakeExecutableTool.readCMakeVersion(CMakeExecutableTool.java:197)
2021-05-27T16:48:57.4731482Z 	at com.jetbrains.cidr.cpp.toolchains.CPPToolchainsPanel$ToolchainPanel$8.readAndCheckVersion(CPPToolchainsPanel.kt:664)
2021-05-27T16:48:57.4732306Z 	at com.jetbrains.cidr.cpp.toolchains.CPPToolchainsPanel$ToolchainPanel$8.readAndCheckVersion(CPPToolchainsPanel.kt:654)
2021-05-27T16:48:57.4733114Z 	at com.jetbrains.cidr.cpp.toolchains.VersionChecker.check(CPPToolchainsUIUtils.kt:191)
2021-05-27T16:48:57.4733973Z 	at com.jetbrains.cidr.cpp.toolchains.CPPToolchainsPanel$ToolchainPanel$addVersionChecker$checkerRunnable$1.runInBackground(CPPToolchainsPanel.kt:903)
2021-05-27T16:48:57.4735044Z 	at com.jetbrains.cidr.cpp.toolchains.CPPToolchainsPanel$ToolchainPanel$addVersionChecker$checkerRunnable$1.runInBackground(CPPToolchainsPanel.kt:896)
2021-05-27T16:48:57.4736054Z 	at com.jetbrains.cidr.cpp.toolchains.CPPToolchainsPanel$ToolchainPanel$addChecker$runnable$1$run$1$backgroundFunction$1.invoke(CPPToolchainsPanel.kt:1121)
2021-05-27T16:48:57.4737066Z 	at com.jetbrains.cidr.cpp.toolchains.CPPToolchainsPanel$ToolchainPanel$addChecker$runnable$1$run$1$backgroundFunction$1.invoke(CPPToolchainsPanel.kt:1095)
2021-05-27T16:48:57.4737935Z 	at com.jetbrains.cidr.cpp.toolchains.CPPToolchainsPanelKt$sam$java_lang_Runnable$0.run(CPPToolchainsPanel.kt)
2021-05-27T16:48:57.4738730Z 	at com.intellij.openapi.progress.impl.CoreProgressManager.registerIndicatorAndRun(CoreProgressManager.java:688)
2021-05-27T16:48:57.4739581Z 	at com.intellij.openapi.progress.impl.CoreProgressManager.executeProcessUnderProgress(CoreProgressManager.java:634)
2021-05-27T16:48:57.4740438Z 	at com.intellij.openapi.progress.impl.ProgressManagerImpl.executeProcessUnderProgress(ProgressManagerImpl.java:64)
2021-05-27T16:48:57.4741332Z 	at com.jetbrains.cidr.cpp.toolchains.CPPToolchainsPanel$ToolchainPanel$addChecker$runnable$1$run$1.run(CPPToolchainsPanel.kt:1122)
2021-05-27T16:48:57.4742226Z 	at com.intellij.util.concurrency.QueueProcessor.runSafely(QueueProcessor.java:238)
2021-05-27T16:48:57.4742889Z 	at com.intellij.util.Alarm$Request.runSafely(Alarm.java:369)
2021-05-27T16:48:57.4743491Z 	at com.intellij.util.Alarm$Request.run(Alarm.java:355)
2021-05-27T16:48:57.4744145Z 	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)
2021-05-27T16:48:57.4744937Z 	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
2021-05-27T16:48:57.4745575Z 	at com.intellij.util.concurrency.SchedulingWrapper$MyScheduledFutureTask.run(SchedulingWrapper.java:220)
2021-05-27T16:48:57.4746248Z 	at com.intellij.util.concurrency.BoundedTaskExecutor.doRun(BoundedTaskExecutor.java:216)
2021-05-27T16:48:57.4747117Z 	at com.intellij.util.concurrency.BoundedTaskExecutor.access$200(BoundedTaskExecutor.java:27)
2021-05-27T16:48:57.4747886Z 	at com.intellij.util.concurrency.BoundedTaskExecutor$1.execute(BoundedTaskExecutor.java:195)
2021-05-27T16:48:57.4748608Z 	at com.intellij.util.ConcurrencyUtil.runUnderThreadName(ConcurrencyUtil.java:213)
2021-05-27T16:48:57.4749330Z 	at com.intellij.util.concurrency.BoundedTaskExecutor$1.run(BoundedTaskExecutor.java:184)
2021-05-27T16:48:57.4750176Z 	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
2021-05-27T16:48:57.4750934Z 	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
2021-05-27T16:48:57.4751672Z 	at java.base/java.util.concurrent.Executors$PrivilegedThreadFactory$1$1.run(Executors.java:668)
2021-05-27T16:48:57.4752427Z 	at java.base/java.util.concurrent.Executors$PrivilegedThreadFactory$1$1.run(Executors.java:665)
2021-05-27T16:48:57.4753099Z 	at java.base/java.security.AccessController.doPrivileged(Native Method)
2021-05-27T16:48:57.4753780Z 	at java.base/java.util.concurrent.Executors$PrivilegedThreadFactory$1.run(Executors.java:665)
2021-05-27T16:48:57.4754455Z 	at java.base/java.lang.Thread.run(Thread.java:829)
2021-05-27T16:48:57.4755824Z Caused by: java.io.IOException: Cannot run program "/home/vsts/work/1/.gradle/caches/modules-2/files-2.1/com.jetbrains.intellij.clion/clion/2021.1/7e6d797e2e496c7163f91c78bb7c6a55725df6da/clion-2021.1/bin/cmake/linux/bin/cmake": error=13, Permission denied
2021-05-27T16:48:57.4757030Z 	at java.base/java.lang.ProcessBuilder.start(ProcessBuilder.java:1128)
2021-05-27T16:48:57.4757741Z 	at java.base/java.lang.ProcessBuilder.start(ProcessBuilder.java:1071)
2021-05-27T16:48:57.4758617Z 	at com.intellij.execution.configurations.GeneralCommandLine.startProcess(GeneralCommandLine.java:458)
2021-05-27T16:48:57.4759420Z 	at com.intellij.execution.configurations.GeneralCommandLine.createProcess(GeneralCommandLine.java:381)
2021-05-27T16:48:57.4760104Z 	... 41 more
2021-05-27T16:48:57.4760600Z Caused by: java.io.IOException: error=13, Permission denied
2021-05-27T16:48:57.4761190Z 	at java.base/java.lang.ProcessImpl.forkAndExec(Native Method)
2021-05-27T16:48:57.4761781Z 	at java.base/java.lang.ProcessImpl.<init>(ProcessImpl.java:340)
2021-05-27T16:48:57.4762400Z 	at java.base/java.lang.ProcessImpl.start(ProcessImpl.java:271)
2021-05-27T16:48:57.4763030Z 	at java.base/java.lang.ProcessBuilder.start(ProcessBuilder.java:1107)
2021-05-27T16:48:57.4763636Z 	... 44 more
```
![image](https://user-images.githubusercontent.com/281596/119959707-8d3c1980-bfa4-11eb-9456-4f66a2d6f689.png)

**Comment by hsz** [comment]
The issue is caused by switching to the programmatic archives extracting. It looks like it doesn't keep the permissions of the files.

**Comment by rli** [comment]
other relevant executables from our CI: https://github.com/aws/aws-toolkit-jetbrains/pull/2598
```
/codebuild/local-cache/custom/e3753f6254b925d6ff833e213fc1705723cd512ae537636977eeb06cc91c3c94/root/.gradle/caches/modules-2/files-2.1/com.jetbrains.intellij.rider/riderRD/2020.2/332d086bc8a2f68cf8cca825116951ec8085dca3/riderRD-2020.2/lib/ReSharperHost/linux-x64/dotnet/dotnet

/codebuild/local-cache/custom/e3753f6254b925d6ff833e213fc1705723cd512ae537636977eeb06cc91c3c94/root/.gradle/caches/modules-2/files-2.1/com.jetbrains.intellij.idea/ideaIC/2020.3/22cef0fc611e4b9642a48bc955b2b7aacb13bd4d/ideaIC-2020.3/bin/linux/fsnotifier64
```

**Comment by hsz** [comment]
Issue is fixed with 7ce6470
